code/workspaces/web-app/src/components/app/AssetRepoSideBar.spec.js

```diff
@@ -1,15 +1,48 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { permissionTypes } from 'common';
+import {
+  renderWithState,
+  buildDefaultTestState,
+} from '../../testUtils/renderWithState';
 import AssetRepoSideBar from './AssetRepoSideBar';
 
+jest.mock('./SideBarButton', () => props => <div {...props}>{props.label}</div>);
+
 const classes = {
   itemList: 'itemList',
   sideBar: 'sideBar',
 };
 
 describe('AssetRepoSideBar', () => {
-  it('renders correctly passing props to children', () => {
+  it('renders correctly when user has no permissions', () => {
+    const props = { classes };
+
+    const state = {
```

Comment:
buildDefaultTestState returns a new object so you don't need to do this

Reply:
Updated now